broadcast/broadcast-client/client_builder_mocks.go

```diff
@@ -0,0 +1,39 @@
+package broadcast_client
+
+import (
+	broadcast_api "github.com/bitcoin-sv/go-broadcast-client/broadcast"
+	"github.com/bitcoin-sv/go-broadcast-client/broadcast/internal/arc/mocks"
+)
+
+type MockType int
+
+const (
+	MockSuccess MockType = iota + 1
+	MockFailure
+	MockTimeout
+	MockAlreadyOnChain
+	MockInMempool
+)
+
+func (cb *builder) WithMockArc(config ArcClientConfig, mockType MockType) *builder {
```

Comment:
Thinking out loud:
As we're providing this as a published component, maybe we should also have tests for it, to ensure that it is working as expected. Beside that those tests can be good intro for anyone who would like to use that in his tests.

Reply:
Sure, good idea, done